types/sanctuary/index.d.ts

```diff
@@ -94,284 +86,493 @@ type Ord<A>
-    //  RegExp
-    regex(flags: RegexFlags): (s: string) => RegExp;
-    regexEscape(s: string): string;
-    test(pattern: RegExp): Predicate<string>;
-    match(pattern: RegExp): (q: string) => Array<Maybe<MatchObj>>;
-    matchAll(pattern: RegExp): (q: string) => Array<MatchObj>;
-    //  String
-    toUpper(s: string): string;
-    toLower(s: string): string;
-    trim(s: string): string;
-    stripPrefix(prefix: string): (q: string) => Maybe<string>;
-    stripSuffix(suffix: string): (q: string) => Maybe<string>;
-    words(s: string): Array<string>;
-    unwords(xs: ReadonlyArray<string>): string;
-    lines(s: string): Array<string>;
-    unlines(xs: ReadonlyArray<string>): string;
-    splitOn(separator: string): (q: string) => Array<string>;
-    splitOnRegex(pattern: RegExp): (q: string) => Array<string>;
+// Maybe
+declare module './TR' {
+  interface TItoTR<A> {
+    'sanctuary-maybe/Maybe@2': Maybe<A>;
   }
+}
+
+export interface Nothing {
+  constructor: MaybeTypeRep;
+}
+
+export interface Just<A> {
+  constructor: MaybeTypeRep;
+  readonly value: A;
+}
+
+export type Maybe<A> = Nothing | Just<A>;
 
-  interface Environment extends Static {
-    //  Configure
-    create(opts: {checkTypes: boolean, env: ReadonlyArray<any>}): Static;
-    env: ReadonlyArray<any>;
-    unchecked: TK;
+export type MaybeTI = 'sanctuary-maybe/Maybe@2';
+
+export interface MaybeTypeRep extends Applicative1<MaybeTI> {
+  <A>(value: A): Maybe<A>;
+  readonly '@@type': MaybeTI;
+}
+
+// Either
+declare module './TR' {
+  interface TItoTR2<L, R> {
+    'sanctuary-either/Either@1': Either<L, R>;
   }
 }
+
+export type EitherTI = 'sanctuary-either/Either@1';
+
+export interface EitherTypeRep extends Applicative2<EitherTI> {
+  <R>(value: R): Either<any, R>;
+  '@@type': EitherTI;
+}
+
+export interface Either<A, B> extends IOrd<Either<A, B>> {
+  constructor: EitherTypeRep;
+}
+
+//  Classify
+export function type(x: any): {
+  namespace: Maybe<string>
+  name: string
+  version: NonNegativeInteger
+};
+export function is(tk: TK): (tk: TK) => TK;
+//  Showable
+export function show(tk: TK): TK;
+//  Fantasy Land
+export function equals<A extends Setoid<any>>(x: A): (y: A) => boolean;
+export function lt<A extends Ord<any>>(x: A): (y: A) => boolean;
+export function lte<A extends Ord<any>>(x: A): (y: A) => boolean;
+export function gt<A extends Ord<any>>(x: A): (y: A) => boolean;
+export function gte<A extends Ord<any>>(x: A): (y: A) => boolean;
+export function min<A extends Ord<any>>(x: A): (y: A) => A;
+export function max<A extends Ord<any>>(x: A): (y: A) => A;
+export function clamp(a: boolean): (b: boolean) => (c: boolean) => boolean;
+export function clamp(a: number): (b: number) => (c: number) => number;
+export function clamp(a: string): (b: string) => (c: string) => string;
+export function clamp<A extends Ord<any>>(a: A): (b: A) => (c: A) => A;
+export function id<A>(p: TypeRep): Fn<A, A> | Category<any>;
```

Comment:
I'm having trouble with this type definition.

`S.id (Function) (42)` gives me this error:

> TS2349: Cannot invoke an expression whose type lacks a call signature. Type `Category<any> | Fn<unknown, unknown>` has no compatible call signatures.

`S.id<number> (Function) (42)` gives me this error:

> TS2349: Cannot invoke an expression whose type lacks a call signature. Type `Category<any> | Fn<number, number>` has no compatible call signatures.

Does the following work?

```suggestion
export function id(p: TypeRep): <A>(a: A) => A | Category<any>;
```

Reply:
I’ll look into it 👏

Reply:
seems to work better